### PR TITLE
ref(ui): icon color type

### PR DESCRIPTION
**Pull request by scttcper**
Change the type of all icon's color property to be be one of the keys of our color theme.

Before you could do `color={theme.gray400}` and that would pass '#gray400' to the icon, now the types only allow for the name `color="gray400"`.

### CODE REVIEW

**Comment by scttcper** [comment]
I'm not totally sure `color="gray400"` is better than `color={theme.gray400}`

File: tests/js/spec/components/sidebar/__snapshots__/index.spec.jsx.snap [hunk 1]
```diff
@@ -6517,11 +6517,11 @@ exports[`Sidebar can have onboarding feature 1`] = `
                                                     onMouseLeave={[Function]}
                                                   >
                                                     <ForwardRef
-                                                      color="#FA5849"
+                                                      color="redLight"
                                                       size="xs"
                                                     >
                                                       <ForwardRef(SvgIcon)
-                                                        color="#FA5849"
+                                                        color="redLight"
```
Comment (EvanPurkhiser) [inline tests/js/spec/components/sidebar/__snapshots__/index.spec.jsx.snap:6524]: Is this right?

Comment (scttcper) [reply tests/js/spec/components/sidebar/__snapshots__/index.spec.jsx.snap:6524]: i think so? shallow snapshot will now show the name of the color being used

File: src/sentry/static/sentry/app/components/emptyStateWarning.tsx [hunk 2]
```diff
@@ -24,7 +23,7 @@ const EmptyStateWarning = ({
   small ? (
     <EmptyMessage className={className}>
       <SmallMessage>
-        {withIcon && <StyledIconWarning color={theme.gray500} size="lg" />}
+        {withIcon && <StyledIconWarning color="gray500" size="lg" />}
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/emptyStateWarning.tsx:26]: I think one benefit of this change is that we don't need to import theme which yields some webpack boilerplate savings.

**Review by markstory — APPROVED** [review]

File: tests/js/spec/components/sidebar/__snapshots__/index.spec.jsx.snap (hunk 1 above)
Comment (priscilawebdev) [reply tests/js/spec/components/sidebar/__snapshots__/index.spec.jsx.snap:6524]: it's correct 😉

**Review by priscilawebdev — APPROVED** [review]
good job! 👏

**Comment by scttcper** [comment]
Change the type of all icon's color property to be be one of the keys of our color theme

**Pull request merged**